### PR TITLE
Enable onephase

### PR DESCRIPTION
**Pull request by hnil**
WIP: Changes to make one phase with and without thermal option work

### CODE REVIEW

File: opm/autodiff/BlackoilModelEbos.hpp [hunk 1]
```diff
@@ -419,6 +419,7 @@ namespace Opm {
 
                 Scalar saturationsOld[FluidSystem::numPhases] = { 0.0 };
                 Scalar oilSaturationOld = 1.0;
+		if(FluidSystem::numActivePhases()>1){
                 if (FluidSystem::phaseIsActive(FluidSystem::waterPhaseIdx)) {
```
Comment (andlaus) [inline opm/autodiff/BlackoilModelEbos.hpp]: please fix the indentation here.

File: opm/autodiff/BlackoilModelEbos.hpp [hunk 2]
```diff
@@ -432,16 +433,20 @@ namespace Opm {
                 if (FluidSystem::phaseIsActive(FluidSystem::oilPhaseIdx)) {
                     saturationsOld[FluidSystem::oilPhaseIdx] = oilSaturationOld;
                 }
-
-                Scalar tmp = pressureNew - pressureOld;
-                resultDelta += tmp*tmp;
-                resultDenom += pressureNew*pressureNew;
```
Comment (andlaus) [inline opm/autodiff/BlackoilModelEbos.hpp]: is this unused? if yes, good catch!

File: opm/autodiff/BlackoilModelEbos.hpp (hunk 1 above)
Comment (hnil) [reply opm/autodiff/BlackoilModelEbos.hpp]: fixed

File: opm/autodiff/BlackoilModelEbos.hpp (hunk 2 above)
Comment (hnil) [reply opm/autodiff/BlackoilModelEbos.hpp]: This code need to be changed, but will change results. I will like that the pull request for facilitating onephase will be merged first. Else a lot of testing has to be done for all relevant cases...

File: opm/autodiff/RateConverter.hpp [hunk 3]
```diff
@@ -493,11 +493,13 @@ namespace Opm {
 
                     // sum p, rs, rv, and T.
                     double hydrocarbonPV = pv_cell*hydrocarbon;
-                    pv += hydrocarbonPV;
-                    p += fs.pressure(FluidSystem::oilPhaseIdx).value()*hydrocarbonPV;
-                    rs += fs.Rs().value()*hydrocarbonPV;
-                    rv += fs.Rv().value()*hydrocarbonPV;
-                    T += fs.temperature(FluidSystem::oilPhaseIdx).value()*hydrocarbonPV;
+		    if(hydrocarbonPV>0){
```
Comment (GitPaean) [inline opm/autodiff/RateConverter.hpp]: Basically here, if there is no hydrocarbon, the quantities will not be averaged.  I guess the code is for the similar cause with https://github.com/OPM/opm-simulators/pull/1961 .

**Comment by atgeirr** [comment]
Replaced by #1969.

**Pull request closed without merging**